### ISSUE TITLE
No longer working

### ISSUE THREAD

**Issue by Charmio1** [NicoSchlueter/Reverse#3]
Firstly I'd like to add my thanks for creating such a brilliant tool. 
It's exactly what I've been looking for however there appears to be an issue.
I've copied the extracted folder "Reverse-master" to "C:\Users\<user>\Documents\Fusion 360\My Add-Ins" and added it into Fusion through the green "+" icon under Utilities > Scripts and Add-ins > Add-Ins (Tab).
The problem is that it doesn't appear to load. The circular icon that shows next to correctly loaded add-ins is missing. There is also nothing showing up when I go to SURFACE > CREATE.

I have tried the usual troubleshooting steps such as restarting Fusion and removing/reinstalling the add-in however this didn't work.
I should say that this is my first time trying to use this add-in so it is possible that it is an issue specific to my setup however I don't believe so. I've also noticed recent comments on the reddit thread from others unable to get it to work, although they don't go into much detail. (https://www.reddit.com/r/Fusion360/comments/id8m7j/i_made_an_addin_for_converting_meshes_to_brep/).

If you have time to attempt to rectify this I would be very grateful.

**Comment by eduarcor** [comment]
Hello Nico, could you please fix it for the 2022 version of fusion360??
Thanks!

**Comment by lucascarvalho** [comment]
Yes, it does not work on MacOS 13.0.1 (22A400) - Fusion 360 2.0.15050 x86_64.

**Comment by bcarriveau** [comment]
Can confirm it does not start, installs just fine but wont run . Im a plus 100 for getting this fixed :-).

**Comment by phossystems** [comment]
Thank you for reporting this issue, and my apologies for ignoring it for so long.

As I suspected this issue is due to the add-in relying on numPy and sciPy for numerical optimisation for fitting the surfaces, and Fusion360 not intending you to, and actively making it difficult to use external libraries.

The hack previously used was to include pre-installed versions of these libraries alongside the script, and adding them to the python path at the start of execution. It appears however the fusion360 python environment has changed so much that the versions shipped with this add-in are no longer compatible.

Updating these libraries could duct tape this project together to run another few months.

Someone more experienced than me could probably find a less hacky way to solve this, similar to this:
https://forums.autodesk.com/t5/fusion-360-api-and-scripts/how-to-use-modules-like-numpy-scipy-etc-by-conda-in-fusion-360/td-p/9355499

I sadly do not have the time to do either.  I have moved on from Fusion360 as I do not agree with the direction Autodesk is heading in.

I'd be happy to accept any pull requests with updated libraries, or hand this project over to a new maintainer if anyone wishes to do so.

My apology

**Comment by Meet-hadvani** [comment]
From my understanding the issue is when Fusion updates python version of fusion also updates and so is it's dependencies. But not all of our Add-in and Script dependencies. 

I resolved it by using os command in python to update those dependencies in whenever Fusion360 version has changed. 

With try catch if import fails means Fusion360 version has changed and in catch update the dependencies to the latest version.

**Comment by lucascarvalho** [comment]
> From my understanding the issue is when Fusion updates python version of fusion also updates and so is it's dependencies. But not all of our Add-in and Script dependencies.
> 
> I resolved it by using os command in python to update those dependencies in whenever Fusion360 version has changed.
> 
> With try catch if import fails means Fusion360 version has changed and in catch update the dependencies to the latest version.

Can you explain how did you do this? Did you manage to get meshes converted into bodies without having thousands of faces, edges and points?
